rules/mapping.go

```diff
@@ -91,6 +91,23 @@ func newMappingRuleSnapshotFromFields(
 	}
 }
 
+func (mrs mappingRuleSnapshot) clone() *mappingRuleSnapshot {
+	filters := make(map[string]string, len(mrs.rawFilters))
+	for k, v := range mrs.rawFilters {
+		filters[k] = v
+	}
+	policies := make([]policy.Policy, len(mrs.policies))
+	copy(policies, mrs.policies)
+	return &mappingRuleSnapshot{
+		name:         mrs.name,
+		tombstoned:   mrs.tombstoned,
+		cutoverNanos: mrs.cutoverNanos,
+		rawFilters:   filters,
+		policies:     policies,
+		filter:       mrs.filter,
```

Comment:
Confirming that you can get away with a ptr copy because the underlying type for mrs.filter is immutable, yeah?